modules/codegen/src/main/scala/com/twilio/guardrail/generators/CirceProtocolGenerator.scala

```diff
@@ -171,11 +171,13 @@ object CirceProtocolGenerator {
 
       case RenderDTOClass(clsName, selfParams, parents) =>
         val discriminators = parents.flatMap(_.discriminators)
-        val parenOpt       = parents.headOption
+        val parentOpt      = parents.headOption
         val terms = (parents.reverse.flatMap(_.params.map(_.term)) ++ selfParams.map(_.term)).filterNot(
           param => discriminators.contains(param.name.value)
         )
-        val code = parenOpt
+
+        val code = parentOpt
+          .filter(p => parents.exists(s => s.discriminators.nonEmpty)) // part of issue222
```

Comment:
This is not the right place to do this filter -- `parentOpt` should not even be able to be constructed. See the suggestion above.